custom/icds_reports/utils/__init__.py

```diff
@@ -84,21 +84,39 @@ class ASRData(object):
 
 class ICDSData(object):
 
-    def __init__(self, domain, filters, report_id):
+    def __init__(self, domain, filters, report_id, override_agg_column=None):
         report_config = ConfigurableReportDataSource.from_spec(
-            self._get_static_report_configuration_without_owner_transform(report_id.format(domain=domain), domain)
+            self._get_static_report_configuration_without_owner_transform(report_id, domain, override_agg_column)
         )
         report_config.set_filter_values(filters)
         self.report_config = report_config
 
-    def _get_static_report_configuration_without_owner_transform(self, report_id, domain):
+    def _get_static_report_configuration_without_owner_transform(self, report_id, domain, override_agg_column):
+        report_id = report_id.format(domain=domain)
         static_report_configuration = StaticReportConfiguration.by_id(report_id, domain)
+
+        if override_agg_column and override_agg_column != 'awc_id':
+            static_report_configuration = self._override_agg(static_report_configuration, override_agg_column)
+
+        # this is explicitly after override, otherwise 'report_columns' attrib gets memoized too early
         for report_column in static_report_configuration.report_columns:
             transform = report_column.transform
             if transform.get('type') == 'custom' and transform.get('custom_type') == 'owner_display':
                 report_column.transform = {}
         return static_report_configuration
 
+    def _override_agg(self, static_report_configuration, override_agg_column):
+        level_order = ['owner_id', 'awc_id', 'supervisor_id', 'block_id', 'district_id', 'state_id']
+        # override aggregation level
+        static_report_configuration.aggregation_columns.remove("owner_id")
```

Comment:
is `owner_id` guaranteed to be in all configurations?

```
>>> columns = []
>>> columns.remove('owner_id')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: list.remove(x): x not in list
```

Reply:
Yes, it should always have it. Was planning to add a validation anyway and missed. Will add.